drivers/hmis/app/models/hmis/ce/match/expression/cde_field_map.rb

```diff
@@ -3,22 +3,52 @@
 module Hmis::Ce::Match::Expression
   # FieldMap implementation for CDE fields
   class CdeFieldMap
+    def initialize(current_date: Date.current)
+      @current_date = current_date
+    end
+
     # Possible reasons why this could this return nil:
     # * the question was left empty on the form
     # * the question was disabled by conditional logic on the form
     # * the version of the form definition did not include the question at the time the form was submitted
-    def instance_value(client, field)
+    def client_query(clients, field)
       cded = parse_entity_type(field)
 
-      cde_values = custom_assessment_cdes(cded, client).
-        map { |cde| cded.read_value_from(cde) }.
-        compact_blank
+      client_ids = clients.pluck(:id)
+      cde_t = Hmis::Hud::CustomDataElement.arel_table
+      values = Hmis::DestinationClientLatestAssessment.
+        where(destination_client_id: client_ids).
+        where(form_identifier: cded.form_definition_identifier).
+        joins(custom_assessment: :custom_data_elements).
+        where(cde_t[:data_element_definition_id].eq(cded.id)).
+        pluck(
+          :destination_client_id,
+          cded.cde_arel_field,
+        )
+
+      result = if cded.repeats?
+        values.group_by(&:first).transform_values { |pairs| pairs.map(&:last) }
+      else
+        values.index_by(&:first).transform_values(&:last)
+      end
```

Comment:
Would it make sense to move this logic into the `cded`?  Something like `result = cded.data_from_values(values)` (or as a class method.)